pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -394,6 +394,12 @@ def _detect_replacable_format_call(self, node: nodes.Const) -> None:
             if "\\" in node.parent.right.as_string():
                 return
 
+            # If % applied to another type than str, it's modulo and can't be replaced by formatting
+            if not hasattr(node.parent.left, "value") or not isinstance(
```

Comment:
In the future I think it is better to test for the class of `left` with an `isinstance`. That's a little more robust and explicit about what we are checking for here.